cli/src/commands/mod.rs

```diff
@@ -2351,6 +2430,134 @@ Please use `jj new 'all:x|y'` instead of `jj new --allow-large-revsets x y`.",
     Ok(())
 }
 
+fn cmd_next(ui: &mut Ui, command: &CommandHelper, args: &NextArgs) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let edit = args.edit;
+    let amount = args.amount;
+    let current_wc_id = workspace_command
+        .get_wc_commit_id()
+        .ok_or_else(|| user_error("This command requires a working copy"))?;
+    let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
+    let current_short = short_commit_hash(current_wc.id());
+    // If we're editing, start at the working-copy commit.
+    // Everything else starts from our direct parent.
+    let start_id = if edit {
+        current_wc_id
+    } else {
+        match current_wc.parent_ids() {
+            [parent_id] => parent_id,
+            _ => return Err(user_error("Cannot run `jj next` on a merge commit")),
+        }
+    };
+    let find_descendant_at = |amount: u64| -> Rc<RevsetExpression> {
+        RevsetExpression::commit(start_id.clone()).descendants_at(amount)
+    };
```

Comment:
nit: Since the closure is called with the same amount in both case, extract a variable for the expression instead, something like this:

```
let descendants_expression = RevsetExpression::commit(start_id.clone()).descendants_at(amount);
```

The same applies to `jj prev`

Reply:
done